### DISCUSSION TITLE
v1.20 feedback thread

### DISCUSSION THREAD

**Discussion by Lailloken** [Lailloken/Exile-UI#18] [Ideas & Feedback]
Share your experience here if you have tested v1.20

**Comment by nebel** [comment]
Was using this version for a few hours tonight at 1200p and noticed no issues. So far I've never had to scan the same mod twice (although this was also true for me in 1.17).

**Comment by noraj** [comment]
Absolutely amazing. I jumped directly for 1.14.1 to 1.20.0.

PS: I think enabling the _Discussion_ Github feature would be appropriate https://docs.github.com/en/discussions

**Comment by leohero92** [comment]
Dear @Lailloken ,

Thank you so much, your tool is like god-level tool to me, really save me this league.
1 suggestion, could you make when click-hold on missing -> auto create regex combined all maps (currently, only map at top) and save the regex into clipboard
![image](https://user-images.githubusercontent.com/6971379/155252952-65faec96-ce59-4fa1-a2ee-1d7ff8f0f8ff.png)

**Reply by Lailloken** [reply]
v1.20.4 has a modification of this feature, hope you like it.

**Reply by leohero92** [reply]
yes yes yes, thank you so much. <3

**Comment by FeralRage7** [comment]
@Lailloken 

Loving the tool and it works great. I'd request two features that may already exist and I just haven't figured it out:

- Save preferred recipes between game sessions: can you write this to some file so when we relaunch the game and the AHK, it puts it back as we last configured it?
- Organize favorites: from what I understand, the ordering matters. When we add favorite recipes, it puts in it's own order (maybe alphabetic)? Can it be added it from left to right as we add the pieces and stay that way?

Thanks for the tool!

**Reply by Lailloken** [reply]
1. The prio-list should be saved whenever the script exits. To test if it works: set up your list, exit the script via the tray icon, open ini\config.ini, and there should be a line "favorite recipes=" followed by what was in the list. If it's not there, tell me. If it's there, but doesn't load on starting the script, also tell me.
2. It is ordered left to right unless one recipe already has an assembled version in the inventory (they are moved to the end of the list then). This is in order to put them lower in the suggestions in case multiple recipes with shared components are ready. The script tries to suggest a route that finishes the whole list the fastest, so unfinished prio-recipes take priority.

**Reply by FeralRage7** [reply]
Thank you for the answers! I do see favorite recipes. 

I'm doing mirror image, treant horde, assassin, rejuvinator recipe and from what I understand, order matters. I think currently I have 3 of the 4 of those done but the incomplete one, rejuvinator, is moved to the front of the list. Are you saying it'll go back to the 4th position (I'm talking about on the top window) once I have acquire the rejuvinator finished piece?

edit: just finished and rejuvinator stays first. And it still shows what pieces are missing? Should this revert to the order I added them?
![wrong order](https://user-images.githubusercontent.com/61394919/155599289-cd4765f3-e4e3-4bdf-918e-1f0f830c42fc.png)

**Reply by Lailloken** [reply]
Did you take the screenshot after scanning?

**Reply by Lailloken** [reply]
If it is after scanning, the missing icons show what is needed to finish the next set. Whenever a recipe is highlighted in green, there's an assembled version in your inventory, and the next one will be started. The tool doesn't stop working on prio-recipes once they're finished, only on sub-components. Your example is a simple priority list, so I can't really explain what the priority system does behind the scenes.
The order also doesn't really matter for your example, but here's how that works: imagine you have components A, B, C; and there are two recipes A+B and B+C. Since you have A, B, C, both recipes are theoretically ready, but you can't do both because after doing one, you would be short one component. Now, whichever recipe is leftmost in your priority list is listed higher in the ready window. That's why the tool automatically positions green recipes at the end, so that other unfinished recipes take priority for shared components. Hope that makes sense.

**Reply by FeralRage7** [reply]
I did take it after scanning and I understand the logic behind the recipes not being ready because by doing one, you use up parts for the other. Thanks for the reply.

**Comment by noraj** [comment]
An auto-update system like in [Poe Trade companion](https://github.com/lemasato/POE-Trades-Companion) would be nice.

**Reply by Lailloken** [reply]
I don't know if that's safe. I don't have any beta testers, so there's a good chance new releases have bugs. This is also a solo project, so there's usually 8-10 hours of absolute support/hotfix downtime. I could think about adding a tooltip popup that indicates that an update is available.

**Reply by noraj** [reply]
Or maybe even if it is manual, have the install/first time launch script create a permanent folder. So that when you launch the new version if will download the files in the permanent folder but keep your config and recognition images. Like the old [PoE Trade macro](https://github.com/PoE-TradeMacro/POE-TradeMacro).

Because manually if I'm at 1.20.4, I have to download 1.20.5 zip, uncompress it, copy 1.20.4 config and recognition images in the 1.20.5 and then change the version number in my shortcut.

**Reply by noraj** [reply]
PS : maybe the upgrade / update system can be copied from either Poe Trade companion or PoE trade macro linked above.

**Reply by Lailloken** [reply]
You can just open the zip, open the Lailloken-UI-1.XX.x folder, and drag the files into the folder on your PC. I made sure the config.ini and recognition files never get overwritten by updates. Unless your system is set up to not integrate the folders correctly, this should work without a problem. There will be an update today, so you can test it with a backed up folder.

**Comment by FeralRage7** [comment]
Hopefully an easy one to implement for next one, can I start the script before starting PoE? I usually launch all my tools and then launch the game. Currently I get a message that the tool can only run once the game has started.

Also with the priority list, can we make it so I can click one of the four and have the tool highlight in the inventory? The only interaction I have with the priority list is long click, which shows ingredients.

**Reply by Lailloken** [reply]
Yes, I can do both. Thanks for your feedback.
